src/NHibernate/QueryOverExtensions.cs

```diff
@@ -0,0 +1,42 @@
+namespace NHibernate
+{
+	// 6.0 TODO: merge into IQueryOver<TRoot>
```

Comment:
Why?

Reply:
For consistency with the few other cases already put in the interface which just delegate to the criteria/root criteria.
We could put instead the converse todo: moving those other options to extension methods.

Reply:
Well interface thing won't work with generic `TQueryOver`.
If you want to stick to interface this change should be reverted and you need also to add public methods to `QueryOver` returning exact type -  to cover detached queryover (see issues like #2177, maybe you should add a test to make sure detached QueryOver actually covered).

But IMHO we should keep them as extensions  - much cleaner implementation.

Reply:
I mean, instead of this PR current todo, the option methods already in the interface could be flagged with the converse todo, like `6.0 TODO: move to extension method`.

Reply:
Yeah I got it and I have no opinion on that... I'm just saying that current TODO is not applicable. Maybe just update this TODO to analyze and move suitable QueryOver interface methods to extensions.

Reply:
Todo changed.

Reply:
> they are quite hard to override.

Yeah but QueryOver is hardwired to criteria and basically wrapper around it. So I don't see it as big drawback here...